dot/types/grandpa_test.go

```diff
@@ -12,6 +12,35 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestEncodeAndDecodeEquivocationPreVote(t *testing.T) {
+	exp := common.MustHexToBytes("0x000a0b0c0d00000000000000000000000000000000000000000000000000000000e7030000010203040000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000506070800000000000000000000000000000000000000000000000000000000") //nolint:lll
```

Comment:
I think we should test that encoding of equivocation produced by substrate can be decoded by us.
Did we get the above value from encoding [sp_finality_grandpa](https://crates.parity.io/sp_finality_grandpa/index.html)::[Equivocation](https://crates.parity.io/sp_finality_grandpa/enum.Equivocation.html#) ?

Reply:
I believe so? I actually wasnt the one to get the bytes, I used it from another test in this file, and since its encoded as a VDT it just pretends a 0

Reply:
@jimjbrettj 
Would nice if you can take that rust struct and encode and and use those bytes here. That way we know that the struct created by us is correct. It would also prevent people from un-necessarily modifying that struct.

Reply:
Yes I will do this, good call